app/javascript/packs/banner_editor.jsx

```diff
@@ -97,6 +99,285 @@ export default class BannerEditor extends React.Component {
+          </div>
+        </div>
+      )
+      case 'editor-banner-image':
+      return(
+        <div>
+          {this.renderControlBar()}
+          <div style={{textAlign:'center', backgroundColor:'white', height:'526px', borderBottomLeftRadius: '8px', borderBottomRightRadius: '8px', paddingTop:'50px', backgroundColor:'rgba(233, 240, 255)'}}>
+          <AvatarEditor
+            ref={this.setLogoEditorRef}
+            image={this.state.tempBannerImage}
+            width={450}
+            height={88}
+            border={50}
+            borderRadius={0}
+            crossOrigin={'anonymous'}
+            color={[233, 240, 255, 0.6]} // RGBA
+            scale={this.state.logoScale}
+            style={{border: '0px solid white'}}
+            rotate={0}
+          />
+          <br/>
+          <input style={{width: '180px', textAlign:'center'}} onChange={ (e) => this.handleZoom(e) } type="range" value={this.state.logoScale} min={1} max={2} step={0.01}/>
+          {/* <img src={this.state.image480}></img> */}
+          </div>
+          {/* <div onClick={ () => this.abc() } className="brave-rewards-banner-control-bar-save-button" style={saveButton}>Crop Image</div> */}
+        </div>
+      )
+      break;
+      case 'editor-logo-added':
+      return(
+        <div>
+          {this.renderControlBar()}
+          <div style={{textAlign:'center', backgroundColor:'white', height:'526px', borderBottomLeftRadius: '8px', borderBottomRightRadius: '8px', paddingTop:'50px', backgroundColor:'rgba(233, 240, 255)'}}>
+          <AvatarEditor
+            ref={this.setLogoEditorRef}
+            image={this.state.tempLogo}
+            width={200}
+            height={200}
+            border={50}
+            borderRadius={100}
+            crossOrigin={'anonymous'}
+            color={[233, 240, 255, 0.6]} // RGBA
+            scale={this.state.logoScale}
+            style={{border: '0px solid white'}}
+            rotate={0}
+          />
+          <br/>
+          <input style={{width: '180px', textAlign:'center'}} onChange={ (e) => this.handleZoom(e) } type="range" value={this.state.logoScale} min={1} max={2} step={0.01}/>
+          {/* <img src={this.state.image480}></img> */}
+          </div>
+          {/* <div onClick={ () => this.abc() } className="brave-rewards-banner-control-bar-save-button" style={saveButton}>Crop Image</div> */}
+        </div>
+      )
+      break;
+    }
+
+    // this.setState({loading:false}, function(){
+    //     // this.setSpinner();
+    //     document.getElementsByClassName("brave-rewards-banner-container")[0].style.width = "600px"
+    // })
+  }
+
+  setLogoEditorRef = (editor) => this.editor = editor
+
+  handleZoom(e){
+    this.setState({logoScale: e.target.value})
+  }
+
+  handleDone(){
+
+    let that = this;
+    let logo = this.state.logo;
+    let bannerImage = this.state.bannerImage;
+    let img = this.editor.getImage();
+
+    switch(this.state.state){
```

Comment:
can we rename the state variable `state` to be something different, i can see this getting confusing fast

Reply:
Any ideas? I chose state to represent the `state` of the banner since there will be several. `editing, logo-editing, banner-editing, preview, notification, etc.`

Reply:
Hmmm, maybe something like `mode`?

Reply:
That's what I was thinking too, I really like state because of its [CS foundations](https://en.wikipedia.org/wiki/State_(computer_science))

But I'll decide on something in the refactor.